pimd/pim_rp.c

```diff
@@ -1135,30 +1135,32 @@ struct pim_rpf *pim_rp_g(struct pim_instance *pim, struct in_addr group)
  * then set the upstream addr as INADDR_ANY and return failure.
  *
  */
-int pim_rp_set_upstream_addr(struct pim_instance *pim, struct in_addr *up,
-			     struct in_addr source, struct in_addr group)
+int pim_rp_set_upstream_addr(struct pim_instance *pim, pim_addr *up,
+			    pim_addr source, pim_addr group)
 {
-	struct rp_info *rp_info;
 	struct prefix g;
 
-	memset(&g, 0, sizeof(g));
-	g.family = AF_INET;
-	g.prefixlen = IPV4_MAX_BITLEN;
-	g.u.prefix4 = group;
+	pim_addr_to_prefix(&g, group);
+
+#if PIM_IPV == 4
+	struct rp_info *rp_info;
 
 	rp_info = pim_rp_find_match_group(pim, &g);
 
 	if (!rp_info || ((pim_rpf_addr_is_inaddr_none(&rp_info->rp))
-			 && (source.s_addr == INADDR_ANY))) {
+			 && pim_addr_is_any(source))) {
 		if (PIM_DEBUG_PIM_NHT_RP)
 			zlog_debug("%s: Received a (*,G) with no RP configured",
 				   __func__);
 		up->s_addr = INADDR_ANY;
 		return 0;
 	}
 
-	*up = (source.s_addr == INADDR_ANY) ? rp_info->rp.rpf_addr.u.prefix4
+	*up = (pim_addr_is_any(source)) ? rp_info->rp.rpf_addr.u.prefix4
 					    : source;
+#else
+	/* TBD */
```

Comment:
I have done this change in my static RP PR.

Reply:
Done